### PR TITLE
[FSSDK-9348] fix: Publish GH Action & other fun additions

### PR DESCRIPTION
**Pull request by mikechu-optimizely**
## Summary
- Fix failing GitHub publish action by separating it and triggering off a new GH Release.
- Add names to workflows, jobs, and steps
- Add missing .github markdown templates for issues and PRs
- Update to nvm version
- Add [devcontainer](https://containers.dev/) configuration so I could run this on Windows

## Test plan
- No code paths changed, so all existing unit tests should continue to pass

## Issues
- [FSSDK-9348](https://jira.sso.episerver.net/browse/FSSDK-9348)

### CODE REVIEW

File: .github/workflows/react.yml [hunk 1]
```diff
@@ -9,46 +9,25 @@ on:
...
     steps:
-    - uses: actions/checkout@v3
+    - name: Checkout branch
+      uses: actions/checkout@v3
     - name: Set up Node ${{ matrix.node }}
       uses: actions/setup-node@v3
       with:
         node-version: ${{ matrix.node }}
     - run: yarn install
     - run: yarn test
```
Comment (mikechu-optimizely) [inline .github/workflows/react.yml:26]: Add `name:` attribute please

File: .github/workflows/react-release.yml
Comment (mikechu-optimizely) [inline .github/workflows/react-release.yml:1]: Rename to react_release.yml

File: .github/workflows/react-release.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    - name: Set up Node
+      uses: actions/setup-node@v3
+      with:
+        node-version: 18
+    - name: Install dependencies
+      run: yarn install
+    - name: Run npm publish command
+      run: npm publish
+      env:
+        NODE_AUTH_TOKEN: ${{ secrets.PUBLISH_REACT_SDK_TO_NPM }}
```
Comment (mikechu-optimizely) [inline .github/workflows/react-release.yml:23]: Move `env` just below `name` in this step.

**Review by mikechu-optimizely — COMMENTED** [review]
Make changes @mikechu-optimizely

**Comment by mikechu-optimizely** [comment]
@rafinutshaw-optimizely I'm ready for another review and will remove @zashraf1985 since he doens't like us any more 😉

**Comment by rafinutshaw-optimizely** [comment]
LGTM.

**Review by rafinutshaw-optimizely — APPROVED** [review]

**Pull request merged**